src/twisted/application/twist/test/test_twist.py

```diff
@@ -191,3 +191,80 @@ def run(self):
             )
         )
         self.assertEqual(runners[0].runs, 1)
+
+
+
+class TwistExitTests(twisted.trial.unittest.TestCase):
+
+    """
+    Tests to verify that the Twist script takes the expected actions related
+    to signals and the reactor.
+    """
+
+    def setUp(self):
+        self.exitWithSignalCalled = False
+
+        def fakeExitWithSignal(sig):
+            """
+            Fake to capture whether L{twisted.application._exitWithSignal
+            was called.
+
+            @param sig: Signal value
+            @type sig: C{int}
+            """
+            self.exitWithSignalCalled = True
+
+        self.patch(_twist, '_exitWithSignal', fakeExitWithSignal)
+
+        def startLogging(_):
+            """
+            Prevent Runner from adding new log observers or other
+            tests outside this module will fail.
+
+            @param _: Unused self param
+            """
+            pass
+
+        self.patch(Runner, 'startLogging', startLogging)
```

Comment:
If memory serves, we don't even have to inject a dummy instance.  I did an interation with a Dummy log factory and the tests will work just fine.  The problem was that downstream logging specific tests failed.  I inferred, from a quick glance at the failure, that those tests were expecting only a certain number of global observers.  We can likely avoid that by manually mutating the global observers if that is possible.  Let me redo it that way and see exactly what those errors were.